File: Lesson09/Timer/app/src/main/java/com/codingblocks/timer/MainActivity.java
package com.codingblocks.timer;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextClock;
import android.widget.TextView;
import android.widget.Toast;

import org.w3c.dom.Text;

import java.net.URI;

public class MainActivity extends AppCompatActivity {

    Button button;

    public String getContentFromUrl(String inputUrl) {
        String output = null;
        //Getting things from the URL
        return output;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        button = findViewById(R.id.btnStart);
        String result = getContentFromUrl("www.google.com");

        if (result != null) {
            Character c = result.charAt(3);
        }

        if (savedInstanceState != null) {
            String key = savedInstanceState.getString("sdfdfs");
        } else {
            Toast.makeText(this, "Bundle is null", Toast.LENGTH_SHORT).show();
        }
//        Toast toast = new Toast(this);
//        toast.setGravity();
//        toast.setMargin();
//        toast.setView();
        final TextView tvStatus = findViewById(R.id.tvStatus);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
//                    Thread.sleep(10000);

                tvStatus.setText("Counting Started");

                Thread t = new Thread(new Runnable() {
                    @Override
                    public void run() {
                        for (long i = 0; i < 1000000L; i++) {
                        }

                        MainActivity.this.runOnUiThread(new Runnable() {
                            @Override
                            public void run() {
                                tvStatus.setText("Counting done!");

                                Toast.makeText(getBaseContext(),
                                        "Counting Done",
                                        Toast.LENGTH_SHORT).show();
                            }
                        });
                    }
                });
                t.start();
            }
        });

    }
}

File: Lesson09/Kotlin/app/src/main/java/com/codingblocks/kotlin/MainActivity.kt
package com.codingblocks.kotlin

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import kotlinx.android.synthetic.main.activity_main.*
import kotlin.concurrent.thread

class MainActivity : AppCompatActivity(), View.OnClickListener {

    private var count = 0
    var note = NoteKt("Test", "27-01-2019")
    val noteFinal = NoteKt("new title")

    fun doSomethingWithInput(str: String) {
        //Do something
    }

    private fun getContentFromUrl(inputUrl: String): String? {
        //Getting things from the URL
        return null
    }

    override fun onClick(v: View?) {
        val id = v?.id
        note.toLowerCase()
        count++
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val result = getContentFromUrl("xhdshj")

        btnTest.setOnClickListener {
            val key = savedInstanceState?.getString("KEY")
            val key2 = savedInstanceState?.getString("KEY2")
        }

        btnTest.setOnClickListener {
            thread {
                //Do something
                runOnUiThread {
                    //Show something on the UI
                }
            }
        }

    }
}